src/main/java/com/rtbhouse/utils/avro/FastDeserializerGenerator.java

```diff
@@ -205,7 +214,21 @@ private void processRecord(JVar schemaVariable, JVar containerVariable, final Sc
                     break;
                 }
             }
-
+            if (!recordAction.getShouldRead()) {
+                return;
+            }
+            // seek through actionIterator also for default values
+            Set<String> fieldNamesSet = record.getFields().stream().map(Schema.Field::name).collect(Collectors.toSet());
+            for (Schema.Field readerField : readerRecord.getFields()) {
+                if (!fieldNamesSet.contains(readerField.name())) {
+                    boolean success = seekToDefault(actionIterator);
```

Comment:
I am not sure if it will ever reach so far. From my experience, if only a new field without default value defined appears, the ResolvingGrammarGenerator will detect such occurrence and will produce an ErrorAction symbol, which will be further interpreted and reported exceptionally by _seekFieldAction_ method.

Reply:
This code (exactly like the next snippet) should just "rewind" action iterator through symbols generated for defaults. It's necessary when old subrecord with new default field is included in schema multiple times. This scenario is checked in FastDeserializerDefaultsTest.shouldAddFieldsInMiddleOfSchema

Note, that for such subrecord all default values are the same and they intend to be like that - they are defined per new field.

Reply:
I agree, the iterator has to be forwarded, but I meant only the section which throws FDGE, in my opinion, it is unnecessary.